### DISCUSSION TITLE
Don't export current value of datasource type dashboard variable

### DISCUSSION THREAD

**Discussion by jangaraj** [grafana/grafana#38256] [Dashboards, Panels & Visualizations]
**What happened**:

Grafana exports (`Export for sharing externally` feature) also current datasource:
```
...
  "templating": {
    "list": [
      {
        "current": {
          "selected": false,
          "text": "my-cloudwatch-datasource",
          "value": "my-cloudwatch-datasource"
        },
        "description": null,
        "error": null,
        "hide": 0,
        "includeAll": false,
        "label": "Datasource",
        "multi": false,
        "name": "datasource",
        "options": [],
        "query": "cloudwatch",
        "queryValue": "hea",
        "refresh": 1,
        "regex": "",
        "skipUrlSync": false,
        "type": "datasource"
      },
...
```

**What you expected to happen**:

Exported current datasource name very likely doesn't exist on another external Grafana instances + it exposes author's Grafana details (datasource name can have a sensitive name), so it should be omitted or parametrized:

```
...
  "templating": {
    "list": [
      {
        "current": {},
        "description": null,
        "error": null,
        "hide": 0,
        "includeAll": false,
        "label": "Datasource",
        "multi": false,
        "name": "datasource",
        "options": [],
        "query": "cloudwatch",
        "queryValue": "hea",
        "refresh": 1,
        "regex": "",
        "skipUrlSync": false,
        "type": "datasource"
      },
...
```

**Environment**:
- Grafana version: 7.5.3
- Data source type & version: NA
- OS Grafana is installed on: NA
- User OS & Browser: NA
- Grafana plugins: NA
- Others: NA

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
